### PR TITLE
Extra requires

### PR DESCRIPTION
**Pull request by jomey**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -139,13 +139,14 @@ def finalize_options(self):
     ],
     extras_require={
         'docs': [
-            'sphinxcontrib-bibtex',
-            'sphinxcontrib-websupport',
-            'pydata-sphinx-theme'
+            'Sphinx>=3.0,<=4.0',
+            'pydata-sphinx-theme',
```
Comment (jomey) [inline setup.py:143]: Will this play nicely with readthedocs? Have no experience with that

File: README.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ git clone https://github.com/USDA-ARS-NWRC/smrf.git
 And install the requirements, SMRF and run the tests.
 
 ```bash
-python3 -m pip install -r requirements_dev.txt
+python3 -m pip install -r requirements_dev.txt .[tests]
```
Comment (jomey) [inline README.md:40]: Forgot to update that with the last PR

File: setup.py (hunk 1 above)
Comment (scotthavens) [reply setup.py:143]: Should, all versions of Sphinx are supported

**Comment by scotthavens** [comment]
Can you add installing the extras into the [docker image](https://github.com/USDA-ARS-NWRC/smrf/blob/dbbaa87583502ceeda23aa1c6f618dfde0958348/Dockerfile#L37)? This also uses nose to test the container was built properly.

File: Dockerfile [hunk 3]
```diff
@@ -34,9 +34,9 @@ COPY --from=builder /root/.local /usr/local
 
 RUN apt-get update -y \
     && apt-get install -y --no-install-recommends libeccodes-tools \
-    && python3 -m pip install --no-cache-dir nose \
+    && python3 -m pip install --no-cache-dir nose smrf[tests] \
     && nosetests -vv --exe smrf \
-    && python3 -m pip uninstall -y nose \
+    && python3 -m pip uninstall -y nose smrf[tests] \
```
Comment (jomey) [inline Dockerfile:39]: I think that should make the Docker built happy again. :crossed_fingers:

Comment (scotthavens) [reply Dockerfile]: This was trying to install smrf from pypi so it didn't. Updated the install to add the extras and the docker builds.

**Review by scotthavens — APPROVED** [review]

**Pull request merged**